### DISCUSSION TITLE
Mocking/Testing with Getters

### DISCUSSION THREAD

**Discussion by marina-mosti** [vuejs/pinia#944] [Help and Questions] [answered]
I'm curious what the best practice is regarding testing complex getters.

Imagine:
If we have a getter that is calling a third file function, some manner of library, that is doing some parsing on an object, or checking a string for a regex. For example, a getter "validEmail" that runs the string through a regex and returns true/false.

When testing this getter on top of the component we have to either:

A. Know what the internal implementation of the getter is, and feed it exactly what we need in order for the getter to behave as intended. This is trivial with a getter like "validEmail" but can be complex when parsing for example a user object, and trying to determine permissions, etc.

B. Mocking the whole module:

```js
import { useStore as useUserStore } from '@/stores/user'

useUserStoremockImplementation(
  () => defineStore('user', {
    state: () => ({
      validEmail: false // replacing getter with state to easily modify it
    })
  })()
)
```

This can be a little cumbersome to do over and over.
Am I missing something obvious?
Is it possible that we could get `createTestingPinia` to also stub out `getters` like it does actions to be able to `mockReturnValue` out of them?

**Comment by posva** [comment, marked as answer]
Opened https://github.com/vuejs/pinia/issues/945

**Reply by marina-mosti** [reply]
Thanks @posva :)
